flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/JsonGenerateUtils.scala

```diff
@@ -227,6 +227,14 @@ object JsonGenerateUtils {
     }
   }
 
+  /**
+   * Determines if the parameter index is % 2, which means checking that we only allow JSON calls
+   * for the VALUE parameter of a JSON_OBJECT call.
+   */
+  def inValuesParam(i: Int): Boolean = {
```

Comment:
I think we can drop this method if we extract the whole `if` condition into a method.

Also about naming: it might make confused some people since in SQL there is also `VALUES` and this method has nothing to do with it

Reply:
I renamed it to inValuesParamOfJsonObject? I think the code is more readable if you read `inValuesParamOfJsonObject` as part of the if condition than `i%2`